### DISCUSSION TITLE
Automatically create server?

### DISCUSSION THREAD

**Discussion by lstefan1520** [gravitational/teleport#11688] [Q&A] [answered]
Hi all,

I am trying to automate and script the process of creating a node on teleport when we run our deployment scripts. 

What’s the best way to do this? Basically I’m trying to get the token that is generated when you click the create server button within the web UI. 

is there a better way to get around my problem?

Any help is appreciated!
Mark

**Comment by webvictim** [comment, marked as answer]
The web UI is basically running `tctl tokens add --type=node --ttl=4h` behind the scenes to get a token. You can copy the `/etc/teleport.yaml` file from a working server, replace the value for `auth_token` with your newly generated token and roll this config out to your other servers (with something else like `cloud-init` or similar handling the installation of Teleport)

You could also just run the `curl` command from the web UI without the `sudo bash` part to get the installation script, save it, edit the embedded join token inside it and then copy and run this script on other machines to install Teleport.

**Reply by lstefan1520** [reply]
Thank you. I appreciate the insight. :)
